allpairspy/allpairs.py

```diff
@@ -141,13 +141,14 @@ def __next__(self):
                 direction = 0
             i += direction
 
-        if len(self.__working_item_matrix) != len(chosen_item_list):
-            raise StopIteration()
-
-        self.__pairs.add_sequence(chosen_item_list)
+            if i == len(self.__working_item_matrix):
+                self.__pairs.add_sequence(chosen_item_list)
+                if len(self.__pairs) == previous_unique_pairs_count:
+                    # could not find new unique pairs - go back to scanning
+                    direction = -1
+                    i += direction
```

Comment:
This fixes insufficient resulting combinations set when filtering is applied.

If `chosen_item_list` doesn't present new pairs, we should get back to scanning until we tried all items. So we will either get new pairs, or brute force all items and [`raise StopIteration()`](https://github.com/thombashi/allpairspy/pull/10/files#R128).